packages/components/pie-toast/src/index.ts

```diff
@@ -1,24 +1,259 @@
+    }
+
+    /**
+     * Render the action button depending on action type and its action.
+     *
+     * @param {ActionProps} action - The action properties.
+     *
+     * @returns {TemplateResult | typeof nothing} - The rendered action button or nothing if the action text is not defined.
+     * @private
+     */
+    private renderActionButton (action: ActionProps) : TemplateResult | typeof nothing {
+        const { text, ariaLabel } = action;
+
+        return html`
+            <pie-button
+                variant="ghost-inverse"
+                size="xsmall"
+                aria-label="${ariaLabel || nothing}"
+                @click="${() => this.handleActionClick()}"
+                class="${componentSelector}-leading-action"
+                data-test-id="${componentSelector}-leading-action"
+                type="button">
+                ${text}
+            </pie-button>
+        `;
+    }
+
+    /**
+     * Template for the footer area.
+     * It should display only when isMultiline is true as well if has action button.
+     * Called within the main render function.
+     *
+     * @private
+     */
+    private renderFooter () {
+        const { leadingAction } = this;
+        return html`
+            <footer class="${componentClass}-footer" data-test-id="${componentSelector}-footer" >
+                ${leadingAction ? this.renderActionButton(leadingAction) : nothing}
+            </footer>
+        `;
+    }
+
+    /**
+     * Template for the close button element. Called within the
+     * main render function.
+     *
+     * @private
+     */
+    private renderCloseButton (): TemplateResult {
+        return html`
+            <pie-icon-button
+                variant="ghost-inverse"
+                size="xsmall"
+                class="${componentClass}-icon-close"
+                data-test-id="${componentSelector}-icon-close"
+                @click="${this.handleCloseButton}">
+                <icon-close></icon-close>
+            </pie-icon-button>`;
+    }
+
+    /**
+     * Template for the toast message. Called within the
+     * main render function.
+     *
+     * @param {string} message - The message to be displayed.
+     * @param {number} messageAreaMaxWidth - The maximum width of the message area calculated in the lifecycle method.
+     *
+     * @private
+     */
+    private renderMessage (message: string, messageAreaMaxWidth: number): TemplateResult {
+        return html`
+            <span style="max-width: ${messageAreaMaxWidth}px" data-test-id="${componentSelector}-message">
+                ${message}
+            </span>
+        `;
+    }
+
+    /**
+     * It handles the action when user clicks in the close button.
+     * Also triggers an event when is executed.
+     *
+     * @private
+     */
+    private handleCloseButton () {
+        this.closeToastComponent();
+        dispatchCustomEvent(this, ON_TOAST_CLOSE_EVENT, { targetNotification: this });
```

Comment:
Do we need this method and closeToastComponent? Or can we just combine the two?

Reply:
@ashleynolan actually no, I was trying to leverage single responsibility here by separating concerns but I guess I went too far 😅 . We can definitely combine it.